satellite/gracefulexit/endpoint.go

```diff
@@ -281,20 +300,17 @@ func (endpoint *Endpoint) doProcess(stream processStream) (err error) {
 
 				exitStatusRequest.ExitSuccess = false
 				// TODO needs signature
-				transferMsg = &pb.SatelliteMessage{
-					Message: &pb.SatelliteMessage_ExitFailed{
-						ExitFailed: &pb.ExitFailed{
-							Reason: pb.ExitFailed_OVERALL_FAILURE_PERCENTAGE_EXCEEDED,
-						},
-					},
+				transferMsg, err = endpoint.getFinishedMessage(ctx, endpoint.signer, nodeID, exitStatusRequest.ExitFinishedAt, exitStatusRequest.ExitSuccess)
+				if err != nil {
+					return rpcstatus.Error(rpcstatus.Internal, err.Error())
 				}
+				transferMsg.GetExitFailed().Reason = pb.ExitFailed_OVERALL_FAILURE_PERCENTAGE_EXCEEDED
```

Comment:
I think changing the exit failed message after signing will make the signature invalid. This should be done in `getFinishedMessage` before the signature.

Reply:
Good catch. Updated. I think I want to add the failure reason to the `nodes` table so we have that to to recreate completed/failed messages. Thoughts?